src/chainstate/stacks/db/blocks.rs

```diff
@@ -4031,14 +4036,17 @@ impl StacksChainState {
         let mut fees = 0u128;
         let mut burns = 0u128;
         let mut receipts = vec![];
+        let mut tx_index: u32 = 0;
```

Comment:
The `tx_index` value refers to the index into the epoch at which this transaction occurs.  One epoch is comprised of the microblock stream that a Stacks miner confirms, plus the anchored block transactions it produces.  So, the value of the first `tx_index` in the receipts for an anchored block should be the one plus the `tx_index` of the last microblock transaction it confirms.  It shouldn't start at 0.  cc @zone117x